src/deepsparse/server/openai_server.py

```diff
@@ -93,49 +93,26 @@ async def create_chat_completion(raw_request: Request):
             request = ChatCompletionRequest(**await raw_request.json())
             _LOGGER.debug("Received chat completion request %s" % request)
 
-            if isinstance(request.messages, str):
-                prompt = request.messages
-            else:
-                # else case assumes a FastChat-compliant dictionary
-                # Fetch a model-specific template from FastChat
-                try:
-                    from fastchat.model.model_adapter import get_conversation_template
-                except ImportError as e:
-                    return create_error_response(
-                        HTTPStatus.FAILED_DEPENDENCY,
-                        f"{str(e)} - Please ensure `fastchat` is installed.",
-                    )
-
-                conv = get_conversation_template(request.model)
-                messages = request.messages
-                messages = messages if isinstance(messages, list) else [messages]
-                # add the model to the Conversation template, based on the given role
-                for message in messages:
-                    msg_role = message["role"]
-                    if msg_role == "system":
-                        conv.system_message = message["content"]
-                    elif msg_role == "user":
-                        conv.append_message(conv.roles[0], message["content"])
-                    elif msg_role == "assistant":
-                        conv.append_message(conv.roles[1], message["content"])
-                    else:
-                        return create_error_response(
-                            HTTPStatus.BAD_REQUEST, "Message role not recognized"
-                        )
-
-                # blank message to start generation
-                conv.append_message(conv.roles[1], None)
-                prompt = conv.get_prompt()
-
-            request_id = f"cmpl-{random_uuid()}"
-            created_time = int(time.time())
             model = request.model
-
             pipeline = app.model_to_pipeline.get(model)
             if not pipeline:
                 return create_error_response(
-                    HTTPStatus.BAD_REQUEST, f"{model} is not available"
+                    HTTPStatus.BAD_REQUEST,
+                    f"The model `{model}` does not exist.",
+                )
+
+            try:
+                prompt = pipeline.tokenizer.apply_chat_template(
+                    conversation=request.messages,
+                    add_generation_prompt=request.add_generation_prompt,
+                    tokenize=False,
```

Comment:
what does this do?

Reply:
It specifies whether the chat template application should add the start of an assistant message to the end of the prompt. vLLM does it by default and this is usually how chat templates end in our examples, so i thought it'd make sense to continue that

for instance here is a tinyllama example from its model card where you can see the `<|im_start|>assistant\n` ending:
```python
from deepsparse import TextGeneration

prompt = "How to make banana bread?"
formatted_prompt =  f"<|im_start|>user\n{prompt}<|im_end|>\n<|im_start|>assistant\n"

model = TextGeneration(model="hf:neuralmagic/TinyLlama-1.1B-Chat-v0.4-pruned50-quant-ds")
print(model(formatted_prompt, max_new_tokens=500).generations[0].text)
```